### ISSUE TITLE
Literal[True] and Literal[False] are not allowed

### ISSUE THREAD

**Issue by alexkoay** [microsoft/pyright#683]
Pyright reports `Literal[True]` and `Literal[False]` as errors, citing that keywords are not allowed in type annotations.

**To reproduce**
`foo: Literal[True] = True` causes an error in the `True` between the square brackets.

**VS Code extension or command-line**
Running pyright 1.1.37 in VS Code on Windows 10.

**Comment by erictraut** [comment]
I'm not able to repro the problem as reported. Pyright supports boolean literals. Here's the full code I'm using.

```
from typing import Literal
foo: Literal[True] = True
```

Are you sure you're using the `Literal` definition from the typing module?

**Comment by alexkoay** [comment]
Oh it might be because I was using it with a blanket import of `typing`.
This is the actual code I've been testing:
```
import typing as t
foo: t.Literal[True] = True
```

**Comment by erictraut** [comment]
Ah yes, I'm able to repro the problem now. "Literal" requires some special-case handling in the parser because the type arguments need to be treated differently from all other classes. My special-case logic in the parser handles the following cases:
* from typing import Literal
* import typing / typing.Literal

But you're using
* import typing as t / t.Literal

I'll make the special-case logic a bit more general to handle this case.

This will be fixed in the next version of Pyright.

In the meantime, you can work around the issue by using one of the supported cases.

**Comment by erictraut** [comment]
This is fixed in version 1.1.39, which I just published.
